tests/model/DataObjectTest.php

```diff
@@ -38,6 +39,13 @@ class DataObjectTest extends SapphireTest {
 	 */
 	public function testSingleton($inst, $defaultValue, $altDefaultValue)
 	{
+		// Calls to scaffold the test database may have cached service specs for DataObjects
+		// with the incorrect 'type' set (singleton instead of prototype)
+		Injector::nest();
+		$reflectionProp = new ReflectionProperty('Injector', 'specs');
+		$reflectionProp->setAccessible(true);
+		$reflectionProp->setValue(Injector::inst(), array());
```

Comment:
This smells, but without adding extra API there’s no way to clear the service spec cache :(